democluster: fix gateway_region() for `--multitenant`

otan opened the pull request:
Resolves #72687

Release note (sql change): Fix `gateway_region` for `--multitenant` demo
clusters.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72734)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

Thanks for getting to this so quickly!

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72734#-:-MoVC4RZDj7L02415EPu:b-as43t9)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan committed:
democluster: fix gateway_region() for `--multitenant`

Release note (sql change): Fix `gateway_region` for `--multitenant` demo
clusters.

otan commented:
bors r=ajstorm
